pkg/supportbundle/spec.go

```diff
@@ -174,8 +173,28 @@ func CreateRenderedSpec(appID string, sequence int64, kotsKinds *kotsutil.KotsKi
 
 // injectDefaults injects the kotsadm default collectors/analyzers in the the support bundle specification.
 func injectDefaults(app *apptypes.App, supportBundle *troubleshootv1beta2.SupportBundle, opts types.TroubleshootOptions, minimalRBACNamespaces []string) error {
-	addDefaultTroubleshoot(supportBundle, app)
+
+	clientset, err := k8sutil.GetClientset()
+	if err != nil {
+		logger.Errorf("Failed to get kubernetes clientset: %v", err)
+	}
+
+	var imageName string
+	var pullSecret *troubleshootv1beta2.ImagePullSecrets
+	if clientset != nil {
+		imageName, pullSecret, err = getImageAndSecret(context.TODO(), clientset)
+		if err != nil {
+			logger.Errorf("Failed to get kotsadm image and secret: %v", err)
+		}
+	}
+
+	addDefaultTroubleshoot(supportBundle)
+	addDefaultDynamicTroubleshoot(supportBundle, app)
 	populateNamespaces(supportBundle, minimalRBACNamespaces)
+	if imageName != "" && pullSecret != nil {
+		populateImages(supportBundle, imageName, pullSecret)
+	}
+	deduplicatedCollectors(supportBundle)
```

Comment:
we should deduplicate analyzers as well if possible

Reply:
thanks, updated